### PR TITLE
Use the actual save function for saving the copy

### PR DESCRIPTION
**Pull request by patrickdemooij9**
Based on the feedback of @KevinJump in https://github.com/umbraco/Umbraco-CMS/pull/11867#issuecomment-1259178819

By using the actual Save function of the service, it'll go through all the needed notifications.

### CODE REVIEW

File: src/Umbraco.Core/Services/DataTypeService.cs [hunk 1]
```diff
@@ -462,7 +462,8 @@ private void ConvertMissingEditorsOfDataTypesToLabels(IEnumerable<IDataType> dat
 
                     copy.Name += " (copy)"; // might not be unique
                     copy.ParentId = containerId;
-                    _dataTypeRepository.Save(copy);
+
+                    Save(copy);
```
Comment (ronaldbarendse) [inline src/Umbraco.Core/Services/DataTypeService.cs:466]: This resets the `CreatorId` to the super user ID (-1), so we should at least keep the existing one:

```suggestion
                    Save(copy, copy.CreatorId);
```

This method should preferably also get the same `userId` parameter, but that might be a breaking change (I'm not sure when this was added to the interface)...

**Review by mikecp — COMMENTED** [review]
Hi @patrickdemooij9 ,

Thank for the update, it's indeed best to have the same behavior on all saves 😉 

However, while trying to figure out if it was a good idea to have a scope defined in the Copy method, and then another scope created in the Save method that would be called within that "copy scope", my attention was brought to the fact that the scope on the Copy method does not seem to be used anywhere. in the method
So, I was wondering if it would not be actually possible to simply remove the scope in Copy all together...
What do you think? Or am I overseeing something?
Cheers!

File: src/Umbraco.Core/Services/DataTypeService.cs (hunk 1 above)
Comment (patrickdemooij9) [reply src/Umbraco.Core/Services/DataTypeService.cs:466]: The Copy interface will be added in 10.3, so does it then still count as a breaking change if it hasn't been released yet?

**Comment by patrickdemooij9** [comment]
Hi @mikecp,

The Scope is actually needed here because of the `_dataTypeContainerRepository.Get(containerId)` call. If I remove the scope, then it doesn't have an ambient scope to use and will throw an exception.
However, I saw that there is an `GetContainer` function that I can use, so that the scope only gets created if we need it.

I also added the change requested by @ronaldbarendse. I am hoping this isn't considered a breaking change as the interface was changed in 10.3 (as long as we push this one before that). Otherwise I'll have to find a way around it

**Comment by patrickdemooij9** [comment]
@mikecp 
Is this something we might still be able to push with Umbraco 10.3? As I am sure @KevinJump will need this for uSync compatibility

**Review by mikecp — APPROVED** [review]

**Comment by mikecp** [comment]
Hi @patrickdemooij9 ,

Thanks for the update, and sorry for the delay. It's all working fine, so let's get this merged!

I'm putting @nul800sebastiaan in here to get his attention on your 10.3.0 question. I hope we can get it in indeed to avoid the breaking change issue.

Cheers!

**Comment by nul800sebastiaan** [comment]
So.. I missed this comment, so this didn't make it to 10.3.0. Looks like this is a breaking change now, which we could add for 11.0.0 but unfortunately it was too late for 10.3. In order to not make it breaking we could do it as a default interface implementation, which would be much preferred since we've also released an RC for v11 already, so new additions like this one will be going in 11.1.0. 

I'll add a PR to fix it so this is non-breaking.

**Comment by nul800sebastiaan** [comment]
Unbreaking in https://github.com/umbraco/Umbraco-CMS/pull/13288

I don't fully understand the scope of the problem, but if we introduced something that breaks uSync, we could call this a regression and add the fix to a 10.3 patch release as well? Does that make sense? // cc @KevinJump

**Comment by KevinJump** [comment]
Hi, 

Yeah, from a uSync (and I guess deploy?) point of view if there are no notifications fired when the copy happens then there is no way to track the changes.

So, if it's gone in to 10.3 without that then - it just means datatypes created by a copy won't be written to disk for syncing (until a save or something else is performed on them by the user).

**Comment by nul800sebastiaan** [comment]
@KevinJump Makes sense!! Would you mind giving this a test run once the build finishes please? Just as a sanity check for me to know if I didn't do anything silly..? 😅 https://github.com/umbraco/Umbraco-CMS/tree/release/10.3.2

**Comment by KevinJump** [comment]
Sure @nul800sebastiaan does it publish the results to a nuget feed anywhere?

**Comment by KevinJump** [comment]
Yeah - seems to work for me from the pre-release feed. 👍

**Comment by nul800sebastiaan** [comment]
Thanks @KevinJump for the super fast turn-around! 🤩

**Pull request merged**